src/main/java/com/couchbase/lite/router/BufferOutputStream.java

```diff
@@ -0,0 +1,31 @@
+package com.couchbase.lite.router;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.LinkedList;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class BufferOutputStream extends OutputStream {
+
+    LinkedList<Byte> buffer = new LinkedList<Byte>();
+    AtomicBoolean closed = new AtomicBoolean(false);
+
+    public void write(int i) throws IOException {
+        synchronized (buffer) {
+            buffer.addLast((byte)i);
+        }
+    }
+
+    LinkedList<Byte> getBuffer() {
+        return buffer;
+    }
+
+    public void close() {
+        closed.set(true);
+    }
+
+    public boolean isClosed() {
+        return closed.get();
+    }
+
```

Comment:
Needs to override [write(byte[], int, int)](http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html#write(byte[],%20int,%20int)) also.